### PR TITLE
Hazard maps

### PR DESCRIPTION
**Pull request by greg-does-weather**
## What does this PR do? 🛠️

Adds maps to alert accordions.

- adds Leaflet with ESRI plugins so we can use the same base map that we use for radar
- adds alert utility method to generate a single combined polygon for any given alert
  - if the alert already has a polygon, just use that
  - if the alert covers one or more zones, fetch their geometries from our database (available since #804), union them into a single geometry, and do geometric simplification to 0.03º resolution
  - if the alert does not have a geometry or any associated zones, that's probably an API bug. but...
    - if the alert has county SAME codes, do the same thing as with zones, except fetch the county geometries first instead of zone geometries
- updates the data layer alert methods to return alerts with full geometries
- updates the alert list twig template to add alert geometries as a data property
- adds new Javascript to load the hazard map
  - hazard maps are two layers: ESRI base map and a GeoJSON polygon
  - hazard maps also get an icon placed at the page's lat/lon
  - hazard maps only attempt to display when:
    - the Leaflet and ESRI libraries have loaded
    - the alert tab is open
    - a specific alert's accordion is open

## What does the reviewer need to know? 🤔

I leaned heavily on @eric-gade's radar loading fixes. High-five for already having solved that!

## Screenshots (if appropriate): 📸

image

### CODE REVIEW

File: web/modules/weather_data/src/Service/SpatialUtility.php [hunk 1]
```diff
@@ -45,4 +45,39 @@ public static function geometryArrayToWKT($geometry)
...
+        // then we are ready to swap them.
+        if (count($point) === 2 && count($numbers) === 2) {
+            return [$point[1], $point[0]];
+        } else {
+            // If every element of the incoming array is itself an array, then
+            // we're ready to recurse on those arrays.
+            $arrays = array_filter($point, "is_array");
+            if (count($point) === count($arrays)) {
+                for ($i = 0; $i < count($point); $i += 1) {
+                    $point[$i] = SpatialUtility::swapLatLon($point[$i]);
```
Comment (eric-gade) [inline web/modules/weather_data/src/Service/SpatialUtility.php:73]: it just recurred to me that you've been _scheme_-ing to do this for a while...

File: web/themes/new_weather_theme/assets/js/components/alertMap.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+        if (expandedNow) {
+          setupMap(alert);
+          observer.disconnect();
+        }
+      });
+      mutationObserver.observe(button, { attributes: true });
+    }
+  }
+};
```
Comment (eric-gade) [inline web/themes/new_weather_theme/assets/js/components/alertMap.js:78]: Would be nice if these accordions dispatched some kind of custom event. Anyway, to my reading this seems functionally equivalent to adding a click handler to the accordion, doing all the stuff, and then removing the handler when it's done the first time. Unless, of course, something besides a click opens the toggle...

File: web/themes/new_weather_theme/assets/js/components/alertMap.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    // keep doing this until we're done.
+    document
+      .querySelector("[data-wx-leaflet]")
+      .addEventListener("load", checkForLeaflet);
+    document
+      .querySelector("[data-wx-leaflet-esri]")
+      .addEventListener("load", checkForLeaflet);
+    document
+      .querySelector("[data-wx-leaflet-esri-vector]")
+      .addEventListener("load", checkForLeaflet);
```
Comment (eric-gade) [inline web/themes/new_weather_theme/assets/js/components/alertMap.js:114]: Nice, never knew about / seen this before

File: web/themes/new_weather_theme/new_weather_theme.libraries.yml [hunk 4]
```diff
@@ -69,6 +69,23 @@ radar:
...
+          crossorigin: /
+  js:
+    https://unpkg.com/leaflet@1.9.4/dist/leaflet.js:
+      data-wx-leaflet: true
+    https://unpkg.com/esri-leaflet@3.0.12/dist/esri-leaflet.js:
+      data-wx-leaflet-esri: true
+    https://unpkg.com/esri-leaflet-vector@4.2.3/dist/esri-leaflet-vector.js:
+      data-wx-leaflet-esri-vector: true
+    assets/js/components/alertMap.js:
+
```
Comment (eric-gade) [inline web/themes/new_weather_theme/new_weather_theme.libraries.yml:88]: It's not strictly necessary, but you _could_ add the `defer` attribute here I think

**Review by eric-gade — APPROVED** [review]
Merge it! 👍   

the lat/lon standards issue == jokes  

## Code Review Checklist

This is an automated comment on every pull request requiring a review. A checked item is either an assertion that I have tested this item or an indication that I have verified it does not apply to this pull request.

## The Basics
- [x] Checks are passing
- [x] I read the code
- [x] I ran the code
- [x] (if applicable) Post deploy steps are run
- [x] (if applicable) I validated the change on the deployed version in <environment>

## Documentation
- [x] changes to “how we do things” are documented in READMEs
- [x] all new functions and methods are commented using plain language
- [x] any new modules added documented in modules.md

## Security
- [x] security false positives are documented
- [x] data from external sources is cleaned and clearly marked

## Reliability
- [x] error handling exists for unusual or missing values
- [x] interactions with external systems are wrapped in try/except
- [x] functionality is tested with unit or integration tests
- [x] dependency updates in composer.json also got changed in composer-lock.json

## Infrastructure
- [x] all changes are auditable and documented via a script
- [x] it is clear who can and should run the script
- [x] (if applicable) diagrams have been updated or added in PlantUML

## Accessibility
- [x] New pages have been added to cypress-axe file so that they will be tested with our automated accessibility testing
- [x] Meets WCAG 2.0 AA or 2.1 AA for Section 508 compliance
    - [ ] Site is keyboard accessible. All interactions can be accessed with a keyboard
    - [ ] Site is free of keyboard traps. The keyboard focus is never trapped in a loop
    - [ ] All form inputs have explicit labels
    - [ ] Form instructions are associated with inputs
    - [ ] All relevant images use an img tag
    - [ ] All images have appropriate alt attributes
    - [ ] Multimedia is tagged. All multimedia has appropriate captioning and audio description
    - [ ] Text has sufficient color contrast. All text has a contrast ratio of 4.5:1 with the background
    - [ ] Site never loses focus. Focus is always visible when moving through the page with the keyboard
    - [ ] Tab order is logical
    - [ ] Tables are coded properly. Tables have proper headers and column attributes
    - [ ] Headings are nested properly. Heading elements are nested in a logical way
    - [ ] Language is set. The language for the page is set
    - [ ] CSS is not required to use the page. The page makes sense with or without CSS
    - [ ] Links are unique and contextual. All links can be understood taken alone, e.g., ‘Read more - about 508’
    - [ ] Page titles are descriptive

## Device Matrix
- [x] firefox/gecko (renders correctly and user interactions work)
- [x] chrome/chromium/edge (renders correctly and user interactions work)
- [x] safari/webkit (renders correctly and user interactions work)
- [x] web page is readable and usable 
    - [ ] at 480px     (mobile)
    - [ ] at 640px     (tablet)
    - [ ] at 1024px    (desktop)

File: web/themes/new_weather_theme/new_weather_theme.libraries.yml (hunk 4 above)
Comment (greg-does-weather) [reply web/themes/new_weather_theme/new_weather_theme.libraries.yml:88]: `defer` unblocks page rendering, right? That's probably a good thing to do!

**Pull request merged**